addons/mail/static/src/models/composer/composer.js

```diff
@@ -682,13 +682,23 @@ function factory(dependencies) {
          * @param {string} mentionKeyword
          */
         _updateSuggestedChannelCommands(mentionKeyword) {
+            const commands = this.env.messaging.commands.filter(
+                command => {
+                    if (!this.thread) {
+                        return false;
+                    }
```

Comment:
This condition is not command specific, no need to be inside the command filter loop. Can directly replace by empty array if there is no thread.